src/serialization/portable/DefaultPortableReader.ts

```diff
@@ -126,59 +133,109 @@ export class DefaultPortableReader implements PortableReader {
         }
     }
 
-    readByteArray(fieldName: string): Buffer {
+    readDecimal(fieldName: string): BigDecimal | null {
+        return this.readNullableField(fieldName, FieldType.DECIMAL, IOUtil.readDecimal);
+    }
+
+    readTime(fieldName: string): LocalTime | null {
+        return this.readNullableField(fieldName, FieldType.TIME, IOUtil.readLocalTime);
+    }
+
+    readDate(fieldName: string): LocalDate | null {
+        return this.readNullableField(fieldName, FieldType.DATE, PortableUtil.readLocalDateForPortable);
+    }
+
+    readTimestamp(fieldName: string): LocalDateTime | null {
+        return this.readNullableField(fieldName, FieldType.TIMESTAMP, PortableUtil.readLocalDatetimeForPortable);
+    }
+
+    readTimestampWithTimezone(fieldName: string): OffsetDateTime | null {
+        return this.readNullableField(fieldName, FieldType.TIMESTAMP_WITH_TIMEZONE, PortableUtil.readOffsetDatetimeForPortable);
+    }
+
+    readByteArray(fieldName: string): Buffer | null {
         const pos = this.positionByField(fieldName, FieldType.BYTE_ARRAY);
+        if (DefaultPortableReader.isNullOrEmpty(pos)) {
+            return null;
+        }
         return this.input.readByteArray(pos);
     }
 
-    readBooleanArray(fieldName: string): boolean[] {
+    readBooleanArray(fieldName: string): boolean[] | null {
         const pos = this.positionByField(fieldName, FieldType.BOOLEAN_ARRAY);
+        if (DefaultPortableReader.isNullOrEmpty(pos)) {
```

Comment:
Can you write a test code that produces `-1` as pos? I couldn't find a way how that might happen. It either throws or returns a valid position

Reply:
Yeah I think you are right, this checks are added quite a while ago in https://github.com/hazelcast/hazelcast/commit/c7b0adc65aa28ae55e43c71a730b62e38c8bf37a. I think we can remove them.

Can you check isNullOrEmpty usages in Java side and verify that it is useless. We should remove it from Java side too.

Reply:
To me, Java side checks also seem unnecessary.